src/cscapi/client.py

```diff
@@ -0,0 +1,276 @@
+            CAPI_METRICS_URL,
+            json={
+                "bouncers": [],
+                "machines": [
+                    {
+                        "last_update": datetime.datetime.now().isoformat(),
+                        "last_push": datetime.datetime.now().isoformat(),
+                        "version": __version__,
+                        "name": machine.machine_id,
+                    }
+                ],
+            },
+        )
+        resp.raise_for_status()
+
+    def _prune_signals(self):
+        signals = filter(lambda signal: signal.sent, self.storage.get_all_signals())
+        self.storage.delete_signals(signals)
+
+    def _refresh_machine_token(self, machine: MachineModel) -> MachineModel:
+        resp = self.http_client.post(
+            CAPI_WATCHER_LOGIN_URL,
+            json={
+                "machine_id": machine.machine_id,
+                "password": machine.password,
+                "scenarios": machine.scenarios.split(","),
+            },
+        )
+        try:
+            resp.raise_for_status()
+        except httpx.HTTPStatusError as exc:
+            logging.error(
+                f"Error while refreshing token: machine_id might be already registered or password is wrong"
+            )
+            raise exc
+
+        new_machine = asdict(machine)
+        new_machine["token"] = resp.json()["token"]
+        new_machine = MachineModel(**new_machine)
+        self.storage.update_or_create_machine(new_machine)
+        return new_machine
+
+    def _register_machine(self, machine: MachineModel) -> MachineModel:
+        logging.info(f"registering machine {machine.machine_id}")
+        machine.password = (
+            machine.password if machine.password else secrets.token_urlsafe(32)
+        )
+        resp = self.http_client.post(
+            CAPI_WATCHER_REGISTER_URL,
+            json={
+                "machine_id": machine.machine_id,
+                "password": machine.password,
+            },
+        )
+        self.storage.update_or_create_machine(machine)
+        return machine
+
+    def _make_machine(self, machine: MachineModel):
+        retrieved_machine = self.storage.get_machine_by_id(machine.machine_id)
+        if not retrieved_machine:
+            machine = self._register_machine(machine)
+        else:
+            machine = retrieved_machine
+        if not machine_token_is_valid(machine):
+            return self._refresh_machine_token(machine)
+        return machine
+
+    def get_decisions(
+        self, main_machine_id: str, scenarios: List[str]
+    ) -> List[ReceivedDecision]:
+        scenarios = ",".join(sorted(set(scenarios)))
+        machine = self._make_machine(
+            MachineModel(machine_id=main_machine_id, scenarios=scenarios)
+        )
+        resp = self.http_client.get(
+            CAPI_DECISIONS_URL, headers={"Authorization": machine.token}
+        )
+
+        return resp.json()
+
+    def enroll_machines(
```

Comment:
|info] I think on relog the token will have info about enroll status can you check with Kevin ? if so we'll create a check enroll status function